lib/ansible/constants.py

```diff
@@ -6,57 +6,53 @@
 __metaclass__ = type
 
 import re
+import warnings
 
 from string import ascii_letters, digits
 
-from ansible.config.manager import ConfigManager
+from ansible.config.manager import ConfigManager, SettingDeprecation
 from ansible.module_utils._text import to_text
-from ansible.module_utils.common.collections import Sequence
 from ansible.module_utils.parsing.convert_bool import BOOLEANS_TRUE
 from ansible.release import __version__
 from ansible.utils.fqcn import add_internal_fqcns
 
 
-def _warning(msg):
+def _warning(warn):
     ''' display is not guaranteed here, nor it being the full class, but try anyways, fallback to sys.stderr.write '''
-    try:
-        from ansible.utils.display import Display
-        Display().warning(msg)
-    except Exception:
-        import sys
-        sys.stderr.write(' [WARNING] %s\n' % (msg))
 
+    depdict = None
+    if isinstance(warn.category, SettingDeprecation):
```

Comment:
```suggestion
    if issubclass(warn.category, SettingDeprecation):
```
warn.category is the class itself, and it doesn't seem like the SettingDeprecation instance is accessible, just used to create the `message`.